HACKING.md

```diff
@@ -0,0 +1,153 @@
+       dd6811c547 UPSTREAM: 78428:Discard stderr output when calling iptables-save
+       d5a5dbeeb5 UPSTREAM: 78428:Better error message if panic occurs during iptables-save output parsing
+       dbf92ac7e1 UPSTREAM: 77303: Update iptables.IsNotFoundError for iptables-nft error messages
+       9bb96ce068 UPSTREAM: <carry>: kube-proxy: make wiring in kubeproxy easy until we sort out config
+       641856db18 (tag: v1.14.0) Merge pull request #75530 from logicalhan/automated-cherry-pick-of-#75529-upstream-release-1.14
+       ...
+
+4. Look through all of the upstream commits on the previous branch:
+   you should be able to drop all of the commits marked `<drop>`, but
+   still need to carry forward all of the commits marked `<carry>`.
+   For the commits tagged with (upstream kubernetes) bug numbers, you
+   need to see if they have been merged in the new branch already or
+   not. (If this is a rebase to a new major release, then they should
+   all have been merged upstream already.)
+
+5. Cherry-pick the commits you need to [the new
+   branch](https://github.com/openshift/kubernetes/commits/sdn-4.3-kubernetes-1.16.0-beta.2):
+
+       00686ab296 (sdn-4.3-kubernetes-1.16.0-beta.2) UPSTREAM: <carry>: Allow low-level tweaking of proxy sync flow
+       d7cd27b93a UPSTREAM: <carry>: kube-proxy: make wiring in kubeproxy easy until we sort out config
+       48ca054dab (tag: v1.16.0-beta.2) Merge remote-tracking branch 'origin/master' into release-1.16
+
+6. Run "make" to ensure it still compiles with the carried patches.
+
+7. Push the new branch to openshift/kubernetes. You should be able to
+   do this directly, without needing a PR.
+
+### Updating the sdn repo dependencies
+
+1. Update the kubernetes dependencies in `glide.yaml`. For
+   `k8s.io/kubernetes` itself that means updating it to point to your
+   new branch. For everything else, it means updating to a branch/tag
+   that matches what you rebased `k8s.io/kubernetes` to. You should not
+   need to create special sdn/openshift-specific branches of anything
+   besides `k8s.io/kubernetes`.
+
+   (For kubernetes dependencies that depend on a commit SHA rather
+   than a branch or tag, see step 2.)
+
+   Occasionally, if you're rebasing to a prerelease version, one of
+   the repos will not have the correct branch/tag yet. In that case,
+   look at the repo, and see if there's another recent tag (eg, from
+   the previous beta) or else see if `master` can reasonably be used.
+   If you do this leave a `# FIXME` comment in `glide.yaml`.
+
+2. Update the remaining kubernetes and openshift dependencies in
+   `glide.yaml`. For dependencies under `github.com/openshift`, and
+   for kubernetes dependencies where we currently depend on a specific
+   commit SHA, see what version is now being used by the origin
+   rebase, and use that. (This is why you waited until after someone
+   started the origin rebase...)
+
+3. Try it out... run
+
+       glide update --strip-vendor
```

Comment:
Ah, I was thinking there should be an automated way to do this and it turns out the answer is `make update-deps`
(@rcarrillocruz )